Sources/Prelude/KeychainClient/KeychainActor.swift

```diff
@@ -0,0 +1,156 @@
+import Foundation
+import KeychainAccess
+
+// MARK: - KeychainActor
+@_spi(KeychainInternal)
+public final actor KeychainActor: GlobalActor {
```

Comment:
@GhenadieVP what do you mean - this has ALWAYS been an actor https://github.com/radixdlt/babylon-wallet-ios/blob/main/Sources/Prelude/KeychainClient/KeychainClient%2BLive.swift#L12

and I have a fix for sync read profile once in app launch in another branch.

Reply:
I know that it has been, but do we want to keep it that way?

Reply:
@GhenadieVP yes ofc! that is what makes it safe! :D otherwise non of the unit tests in KeychainTests would pass...

Reply:
@GhenadieVP right, so, we could need to SPLIT `KeychainActor` and thus `KeychainClient` into two different sets for that to happen. Since an actor *cannot let a stored property be marked `nonisolate`*, and in order to let `reading without auth` be non async, the methods would need to be `nonisolated`, which is not possible since they access actor-isolated `let keychain` stored property.

Reply:
Are we only limited to using Actors? We cannot have the same functionality without needing to have a KeychainActor?

Reply:
Actor is a great thing! it is what protects us from bugs in Keychain, at least after this PR, which does not use `Task {}` **unstructured** concurrency, like we had before, which was wrong.

Without actors, we can accidentally corrupt keychain data. We use `async` functions all over the code anyway, I don't really understand the reluctance? 

In a coming PR I'm gonna refactor ProfileStore, to be much simpler, using a single read of `profileSnapshot: ProfileSnapshot?` part of init of KeychainActor, which I hope will allow ProfileStore init to be sync.

Reply:
I mean, if we would only have ONE SINGLE ACTOR in the whole code base, I would let it be the KeychainActor. It is the only crucial data we have in our entire code base.

Reply:
The problem with async is that it breaks TCA reducers. Instead of just reading from the keychain and going on with your day, you have to send an internal action, pick up that action and then continue doing what you did. So we should ideally limit async methods in clients as much as possible.

Would it not be possible to for example have an actor for the auth requiring methods, inside a type that is itself not an actor?

Reply:
Lets take the actor discussion on Slack/on a call, since it does not related to this PR, since this PR does not change that... only moved the existing actor out into a seperate file.